prusti-viper/src/encoder/high/procedures/inference/semantics.rs

```diff
@@ -219,6 +232,18 @@ impl CollectPermissionChanges for vir_high::Rvalue {
     }
 }
 
+impl CollectPermissionChanges for vir_high::ast::rvalue::Ref {
+    fn collect(
+        &self,
+        consumed_permissions: &mut Vec<Permission>,
+        produced_permissions: &mut Vec<Permission>,
+    ) -> SpannedEncodingResult<()> {
+        consumed_permissions.push(Permission::Owned(self.place.clone()));
+        produced_permissions.push(Permission::Owned(self.place.clone()));
```

Comment:
Shouldn't this add `SharedBorrowed`/`MutBorrowed` instead of `Owned`?

Reply:
Looking into it, it seems to involve a cascade of changes.

Reply:
Yes, that is expected. Let's merge first with `Owned` and fix this in a later PR.

Reply:
I already added them :)

Reply:
undid this as discussed.